File: 面试准备第四天/面试准备第四天/面试准备第四天/ViewController.m
//
//  ViewController.m
//  面试准备第四天
//
//  Created by YoungEE on 2020/7/6.
//  Copyright © 2020 YoungEE. All rights reserved.
//

#import "ViewController.h"
#import "lei.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
//    lei *l = [lei new];
    // Do any additional setup after loading the view.
}

-(void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
    
    NSMutableArray *array = [NSMutableArray arrayWithCapacity:4];
    [array addObject:@"1"];
    [array addObject:@"2"];
    [array addObject:@"3"];
    [array addObject:@"4"];
    NSLog(@"%@",array);
    array = [NSMutableArray arrayWithCapacity:8];
    [array addObject:@"5"];
    [array addObject:@"6"];
    [array addObject:@"7"];
    [array addObject:@"8"];
    NSLog(@"%@",array);
}


@end

File: 算法/面试demo/面试demo/ViewController.swift
//
//  ViewController.swift
//  面试demo
//
//  Created by YoungEE on 2020/7/29.
//  Copyright © 2020 YoungEE. All rights reserved.
//

import UIKit

//给一个有序数组,如[1,1,2,2,3,4,5,5],使用 O(1) 额外空间,要求每个数字只出现一次，并返回数组新长度

class ViewController: UIViewController {
    let nums = [1,1,2,2,3,4,5,5]
    

    override func viewDidLoad() {
        super.viewDidLoad()
        print(self.method())
        //买彩票 总共有三个盒子只有一个盒子有奖品 随便选一个的概率是1/3概率，现在我选一个，另外一个人选一个，售票员选一个，那么这个时候每个人的概率是1/3，这时候售票员打开一个盒子没有奖品，那么我要不要和另外一个人换？
        //我在选一个现在两个 其中一个打开  里面没有奖品
        //
        // Do any additional setup after loading the view.
    }
  
    func method() -> Set<Int>{
//        var temp = [Int]()
        //定义一个集合来装载数据
        var temp1 = Set<Int>()
        //遍历当前数组
        for (_ , val) in nums.enumerated(){
            //如果集合中未包含当前的数字加入到结合当中，由当前
            if !temp1.contains(val) {
                temp1.insert(val);
            }
        }
        return temp1
    }

}

